### PR TITLE
update of the arch.md document

### PR DESCRIPTION
**Pull request by tremes**
This updates the `docs/arch.md` document. It's still not a masterpiece, but it's improving. This PR:
- describes all the config options in the `support` secret
- updates the clusteroperator conditions & status section
- updates, simplifies and reword some important sections

**The PR https://github.com/openshift/insights-operator/pull/621 should be merged first!!!**

## Categories

- [ ] Bugfix
- [ ] Enhancement
- [ ] Backporting
- [X] Others (CI, Infrastructure, Documentation)

## Sample Archive

No new data

## Documentation

Major update of the `docs/arch.md` document

## Unit Tests

No new test

## Privacy

Yes. There are no sensitive data in the newly collected information.

## Breaking Changes

No

## References

https://issues.redhat.com/browse/???
https://bugzilla.redhat.com/show_bug.cgi?id=???
https://access.redhat.com/solutions/???

### CODE REVIEW

File: docs/arch.md [hunk 1]
```diff
@@ -154,47 +166,100 @@ oc get co insights -o=json | jq '.status.conditions'
...
+    "message": "Insights works as expected",
+    "reason": "AsExpected",
+    "status": "True",
+    "type": "Available"
   }
 ]
 ```
+A condition is defined by its type. You may notice that there are some non-standard clusteroperator conditions. They are:
+- `SCAAvailable` - based on the SCA (Simple Content Access) controller in `pkg/ocm/sca/sca.go` and provides information about the status of downloading the SCA entitlements.
+- `ClusterTransferAvailable` - based on the cluster transfer controller in `pkg/ocm/clustertransfer/cluster_transfer.go` nad provides information about the availability of cluster transfers.
```
Comment (JoaoFula) [inline docs/arch.md:220]:
```suggestion
- `ClusterTransferAvailable` - based on the cluster transfer controller in `pkg/ocm/clustertransfer/cluster_transfer.go` and provides information about the availability of cluster transfers.
```

**Review by rluders — APPROVED** [review]
I just reviewed all changes. Looks good.

**Pull request merged**